CMakeLists.txt

```diff
@@ -0,0 +1,1042 @@
+foreach(loop IN ITEMS ${ASTUCE_MASKS})
+	add_custom_command(TARGET copy_input_files POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${loop} ${CMAKE_BINARY_DIR}/ast_mask/ DEPENDS ast_mkdir)
+endforeach()
+
+#We need to build asatuce and dafor before we build SixTrack, but crlibm and so on can be built at the same time.
+add_dependencies(SixTrack copy_input_files generate_input)
+
+###################################################################################################
+#running astuce and dafor
+###################################################################################################
+add_custom_target(run_astuce)
+
+#make running astuce depend on astuce being built and the input files being in place
+add_dependencies(run_astuce astuce astuce_clean_define_flags astuce_clean_decks)
+
+#This target cleans the define flags
+add_custom_target(astuce_clean_define_flags)
+#this requires the input files to be in place
+add_dependencies(astuce_clean_define_flags copy_input_files)
+
+#this target cleans the decks
+add_custom_target(astuce_clean_decks)
+#and requires the define flags to already be cleaned (this is just to stop parallel writes on the same file)
+add_dependencies(astuce_clean_decks astuce_clean_define_flags)
+
+#This macro calls the regex in masks_define_flags.cmake to remove the requested flags
+macro(ast_masks OPT FILE_NAME)
+	add_custom_command(TARGET astuce_clean_define_flags POST_BUILD COMMAND ${CMAKE_COMMAND} ARGS -DFILE_NAME:FILEPATH=${CMAKE_BINARY_DIR}/ast_mask/${FILE_NAME}.ast -DOPT:STRING=${OPT} -P ${CMAKE_SOURCE_DIR}/SixTrack/masks_define_flags.cmake)
+endmacro()
+
+#This macro calls the regex in masks_decks.cmake to remove the requested decks
+macro(ast_masks2 OPT FILE_NAME)
+	add_custom_command(TARGET astuce_clean_decks POST_BUILD COMMAND ${CMAKE_COMMAND} ARGS -DFILE_NAME:FILEPATH=${CMAKE_BINARY_DIR}/ast_mask/${FILE_NAME}.ast -DOPT:STRING=${OPT} -P ${CMAKE_SOURCE_DIR}/SixTrack/masks_decks.cmake)
+endmacro()
+
+#First clean up the astuce decks for only what is enabled
+#set(OPT crlibm tilt fast bpm beamgas collimat bnlelens cr boinc nagfor bignblz debug hdf5 fio lf95 hugenblz stf bignpart datamods hugenpart)
+foreach(loop IN ITEMS ${FORS})
+	if(NOT CRLIBM)
+		ast_masks(crlibm ${loop})
+	endif()
+	if(NOT TILT)
+		ast_masks(tilt ${loop})
+	endif()
+	if(NOT FAST)
+		ast_masks(fast ${loop})
+	endif()
+	if(NOT BPM)
+		ast_masks(bpm ${loop})
+	endif()
+	if(NOT BEAMGAS)
+		ast_masks(beamgas ${loop})
+	endif()
+	if(NOT COLLIMAT)
+		ast_masks(collimat ${loop})
+	endif()
+	if(NOT BNLELENS)
+		ast_masks(bnlelens ${loop})
+	endif()
+	if(NOT CR)
+		ast_masks(cr ${loop})
+	endif()
+	if(NOT BOINC)
+		ast_masks(boinc ${loop})
+	endif()
+	if(NOT NAGFOR)
+		ast_masks(nagfor ${loop})
+	endif()
+	if(NOT BIGNBLZ)
+		ast_masks(bignblz ${loop})
+	endif()
+	if(NOT DEBUG)
+		ast_masks(debug ${loop})
+	endif()
+	if(NOT HDF5)
+		ast_masks(hdf5 ${loop})
+	endif()
+	if(NOT FIO)
+		ast_masks(fio ${loop})
+	endif()
+	if(NOT LF95)
```

Comment:
There is a nagfor flag in this gigantic IF, a few lines above.

Leave the LF95 flag for now, it may come back (you've seen the emails) - apparently there is a newer version. The pre-2003 version is dead.